Make new build default and combine into dist package (#3411)
NVIDIA/spark-rapids: CONTRIBUTING.md

```

After a successful build the RAPIDS Accelerator jar will be in the `dist/target/` directory.
This will build the plugin for a single version of Spark.  By default this is Apache Spark
3.0.1. To get an uber jar with more then 1 version you have to `mvn install` each version

> and then use one of the defined profiles in the dist module.

Author: need to add more options here

```

After a successful build the RAPIDS Accelerator jar will be in the `dist/target/` directory.
This will build the plugin for a single version of Spark.  By default this is Apache Spark
3.0.1. To build against other versions of Spark you use the `-Dbuildver=XXX` command line option
to maven. For instance to build Spark 3.1.1 you would use:

```shell script
mvn -Dbuildver=311 verify
```
You can find all available build versions in the top level pom.xml file. Currently
they include 301, 302, 303, 304, 311, 312, 313, 320 and 311cdh. If you are building

> Databricks suggest you use the `jenkins/databricks/build.sh` script and modify it for

Reviewer: ```suggestion
for Databricks then you should use the `jenkins/databricks/build.sh` script and modify it for
```

> To get an uber jar with more then 1 version you have to `mvn install` each version

Reviewer 2: ?
```suggestion
To get an uber jar with more than 1 version you have to `mvn install` each version
```

To get an uber jar with more then 1 version you have to `mvn install` each version
and then use one of the defined profiles in the dist module. See the next section
for more details.

> ### Building a distribution for multiple versions of Spark

Reviewer: Nit: Capitalization is not consistent with other headings.
```suggestion
### Building a Distribution for Multiple Versions of Spark
```

By default the distribution jar only includes code for a single version of Spark. If you want
to create a jar with multiple versions we currently have 4 options.

> 1. Build for all Apache Spark versions and CDH with no SNAPSHOT versions of Spark, only released. Use `-Pallnosnapshots`.

Author: suggestions on naming conventions here?  I realize its a bit weird with allno or all nowithDB, etc..

Reviewer 3: It cannot be all if it does not include databricks.  I would replace all with default, standard, or just drop it all together.

`-PnoSnapshots`
`-Psnapshots`
`-PnoSnaphsotsWithDatabricks`
`-PsnapshotsWithDatabricks`

Also we should have a way to distinguish words in the profiles. snake_case is good, but I don't know what convention (if any) maven has or that we have used in our poms.

2. Build for all Apache Spark versions and CDH including SNAPSHOT versions of Spark we have supported for. Use `-Pallwithsnapshots`.
3. Build for all Apache Spark versions, CDH and Databricks with no SNAPSHOT versions of Spark, only released. Use `-PallnosnapshotswithDB`.
4. Build for all Apache Spark versions, CDH and Databricks including SNAPSHOT versions of Spark we have supported for. Use `-PallwithsnapshotsDB`

You must first build and install each of the versions of Spark and then build one final time using the profile for the option you want.

For instance if I want all Apache Spark versions and CDH with no SNAPSHOT versions I would run something like this:

```shell script

> mvn -U -Dbuildver=301 clean install -Drat.skip=true -DskipTests

Reviewer: Where does the rat check run in this series of build commands?  Should we not be skipping it on the first build?

> mvn -Dbuildver=301 clean install -DskipTests

Reviewer 2: Is `-Drat.skip=true` not needed for building 301 ?

Author: we want it to run with one of the builds

> mvn -U -Dbuildver=302 clean install -Drat.skip=true -DskipTests

Reviewer: Is the `-U` necessary?  Seems like this is only needed when building for snapshots, and that doesn't seem to apply for these commands.

mvn -U -Dbuildver=303 clean install -Drat.skip=true -DskipTests
mvn -U -Dbuildver=311 clean install -Drat.skip=true -DskipTests
mvn -U -Dbuildver=312 clean install -Drat.skip=true -DskipTests
mvn -U -Dbuildver=311cdh clean install -Drat.skip=true -DskipTests
mvn -U -pl dist -Pallnosnapshots package -DskipTests
```

> There is also a build script `build/buildall` to build everything with snapshots and this will have more options to build later.

Reviewer: Shouldn't we lead this section with the pointer to the script and go into nitty-gritty details after?  We want to point people to the thing we want them to do first.

I guess `buildall` doesn't support building without snapshots and that's why?  Seems like we'd want to fix that pronto.

Author: yeah we can rearrange and yes want to expand buildall very quickly after this
